src/redux/gatekeeperText.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+import type { ReduxMiddleware } from './store';
+
+function nameTemplate(fullText, playerName, bossName) {
+  return fullText.replace(/\[Name\]/g, playerName).replace(/\[BOSS\]/g, bossName);
+}
+
+const gatekeeperTextMiddleware: ReduxMiddleware = (store) => (next) => (action) => {
+  const result = next(action);
+  const currentState = store.getState();
+
+  if (
+    currentState.stage === 'fight' &&
+    ['perform_player_attack', 'perform_boss_attack'].includes(action.type)
+  ) {
+    if (currentState.attack) {
```

Comment:
Do we need all of this or can it just be `if (currentState.stage === 'fight' && currentState.attack)`?